specifyweb/frontend/js_src/lib/components/WorkBench/WbSpreadsheet.tsx

```diff
@@ -259,32 +227,66 @@ function WbSpreadsheetComponent({
         columns={columns}
         commentedCellClassName="htCommentCell"
         comments={comments}
-        data={data as (string | null)[][]}
+        contextMenu={contextMenuConfig}
+        // @ts-expect-error data throws error for readonly input
+        data={data as readonly (readonly (string | null)[])[]}
```

Comment:
hmm, previously we had a type assetion to a mutable type here for that reason

could you check if we are mutating `data` internally?
if so, we should change our typescript type to not be readonly
if we are not mutating it internally (and HOT does all of the mutation), then we should have a type assertion to a mutable type here (`(string | null)[][]` like it was before, or something like that)

it's usually better to have a type assertion in TypeScript than silence an error
and usually better to have the weakest type assertion you can

Reply:
The linter kept making it readonly and I initially couldn't figure out how to disable it for that line. I have added an eslint-disable now. Also, hot doesn't accept a readonly array for that prop

Reply:
> The linter kept making it readonly

yes, it's a bit overly eager 😊

most of the time it's good that linter defaults everything to readonly
but when there are exceptions, you can do one of these:
- use `WritableArray` - it's just like `[]`, but doesn't trigger the linter
  - you may say linter adds extra overhead. somewhat true - the goal is to make readonly the easier default. and mutable only as an opt-in if you really need to
- use `mutable()` function - casts any value into mutable type (shallow-cast only, and affects types only, not runtime)
- silence eslint